microprofile/WFLY-19591_MicroProfile_OpenAPI_4.0.adoc

```diff
@@ -0,0 +1,138 @@
+// Specify the Feature Development tracker issue for the feature.
+// This must be an issue tracked in https://github.com/orgs/wildfly/projects/7/views/1.
+// To create a Feature Development tracker issue, go to  https://github.com/wildfly/wildfly-proposals/issues/new/choose
+// and select 'Feature Development'
+issue: https://github.com/wildfly/wildfly-proposals/issues/647
+// Provide the github ids of the members of the feature team, organized by role.
+// Provide a single id for the 'assignee' role. Use a yaml list for the 'sme' and
+// 'outside-perspective' roles, even if there is only one person in a role.
+feature-team:
+ developer: pferraro
+ sme: fburgizo
+  -
+ outside-perspective:
+  -
+// If this issue tracks the promotion to a higher stability level of a previously
+// completed feature, provide the URL of the https://github.com/wildfly/wildfly-proposals/issues
+// issue that was used to track the previous feature.
+promotes:
+// This should be blank during initial development of a feature. It may be used
+// after the feature is completed if a subsequent issue is field to track promotion
+// of this feature to a higher stability level
+promoted-by:
+---
+= MicroProfile OpenAPI 4.0
+:author:            Paul Ferraro
+:email:             paul.ferraro@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WildFly currently provides MicroProfile OpenAPI 3.1 support to applications.
+This feature request upgrades support for this specification to version 4.0 as part of version 7.0 of the MicroProfile platform specification.
+
+Most notably, this release of the MicroProfile OpenAPI specification generates documentation using version 3.1 of the OpenAPI specification.
+
+https://download.eclipse.org/microprofile/microprofile-open-api-4.0/microprofile-openapi-spec-4.0.html#release_notes_40
+
+=== User Stories
+
+This is feature is covered by existing user stories.
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/WFLY-19591[Parent issue]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19588
+
+=== Affected Projects or Components
+
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Upgrade version of MicroProfile OpenAPI specification API to 4.0.
+* Upgrade version of SmallRye OpenAPI to 4.0.x
+
+=== Changed requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+=== Future Work
+
+N/A
+
+== Backwards Compatibility
+
+=== Default Configuration
+
+N/A
+
+=== Importing Existing Configuration
+
+N/A
+
+=== Deployments
+
+See https://download.eclipse.org/microprofile/microprofile-open-api-4.0/microprofile-openapi-spec-4.0.html#incompatible_changes_40[here] for the list of incompatible changes between versions 3.1 and 4.0 of the MicroProfile OpenAPI specification.
```

Comment:
> I am wondering what this means, actually, for users, and how it could be mitigated.

It means that user deployments may need to be updated.  There is no mitigation.

> Would existing deployments, specifically those carrying static OpenAPI documents, work seamlessly?

These need to be valid OpenAPI 3.1 documents (see 1st bullet point in link).

> I am also thinking at users/customers that have automation in place that consumes the generated OpenAPI documentation, and what would happen to such components, although this is probably out of scope here. Not sure, WDYT?

That is not in scope.

Reply:
Thanks for your feedback. Everything here makes sense but will have implications on the existing test coverage, and changes to the existing documentation.

Reply:
> Everything here makes sense but will have implications on the existing test coverage,

Function test coverage is the province of the new version of the MP OpenAPI TCK.

> , and changes to the existing documentation.

Documentation changes are limited to version number changes.